src/main/java/transforms/SplitTransform.java

```diff
@@ -72,19 +74,11 @@ public Object applyValue( String o ) {
             idx = parseInt( index );
         }
 
-        if ( !( ifMoreThan == null ) ) {
-            if ( !( ifMoreThan < strNames.length ) ) {
-                return null;
-            }
-        }
-
-        if ( strNames.length > idx ) {
+         if ( strNames.length > ifMoreThan.orElse(idx ) ) {
             return strNames[ idx ].trim();
         }
-        if ( !StringUtils.isBlank( valueElse ) ) {
-            return valueElse;
-        }
-        return null;
+
+        return this.valueElse.orElse(null);
```

Comment:
I think this removes the check for empty strings so if `valueElse` is `Optional.of("")`, it would not get returned as `null` anymore. Is that the intention?

Reply:
I believe `null` and `""` go into the db as the same.

Reply:
They don't -- there might be an extra step somewhere in new shuttle that converts empty strings to nulls, but empty string is an actual value that's written (e.g. if you write an empty string to nc.PersonMiddleName, when you read the entity, you'll see `{ "nc.PersonMiddleName": [""] }` but if you write null to nc.PersonMiddleName you'll see `{ "nc.PersonMiddleName": [] }`

Reply:
Oh interesting !
In that case: if valueElse is indeed `""`, we might want to keep that indeed as `""`?

Reply:
Oh interesting !
Yeah, shuttle replaces in a lot of cases `""` with `null`, and data science team agrees on doing that, but I don't think it's consistent throughout all of shuttle.
So in this case, we should indeed put the check back, but shouldn't we then rather put it somewhere [here](https://github.com/openlattice/shuttle/blob/develop/src/main/java/com/openlattice/shuttle/transformations/Transformation.java#L98) so that it applies for all of shuttle?

Reply:
(a) Wouldn't it go in getInputString? 
(b) I don't think Shuttle should do anything really-- it should pass the underlying raw data value (null or "") to the transformations.

Reply:
It's not just in getInputString, because it might be "" after transformation...

Reply:
Will it ever be null after transformation?

Reply:
Yeah ! After cleaning.  
For example when we parse a name, but there's no MiddleName present...